### ISSUE TITLE
Download along linestring extension for (around: ) filter

### ISSUE THREAD

**Issue by mmd-osm** [drolbr/Overpass-API#418]
This is a proposed extension for the `around` filter to enable a download along a linestring scenario, rather than only a single lat/lon. The sequence of lat/lon pairs is handled like one virtual way, to query for all objects within a given radius to this way. This is extremely handy to download data along a list of GPS coordinates without the hassle of calculating some suitable polygon for `(poly: ...)` upfront.

### Closed way

http://overpass-turbo.eu/s/qX9

![grafik](https://user-images.githubusercontent.com/5842757/29240646-95dbbd7c-7f69-11e7-91f5-002debd6d97b.png)

```
way[highway]
(around:500,50.540853270068986,8.048780365649707,50.53106288705902,8.030823236553783,
50.51780737956311,8.019643105996508,50.50287491071276,8.016749912560886,
50.48828159051387,8.022534398052139,50.47599950382573,8.036215335651725,
50.467689755650376,8.055945038928135,50.46447690759688,8.079058902127825,
50.46679590711731,8.102435269947343,50.47433280529453,8.122917034379736,
50.48606755168466,8.137738019645033,50.50041288059356,8.144896569557243,
50.51542994506574,8.143425882283827,50.529090915610794,8.13352458229042,
50.53955268865336,8.11652989500613,50.545404823255616,8.09473704711951,
50.545858734919165,8.07108928349599,50.540853270068986,8.048780365649707)   
;(._;>;);out meta;
```

### Road trip

http://overpass-turbo.eu/s/qXI

-> Produces 10 MB PBF file, imported via osm2pgsql, rendered via Kosmtik

![grafik](https://user-images.githubusercontent.com/5842757/29248519-61e1cf6e-801a-11e7-8256-87917b134971.png)

![grafik](https://user-images.githubusercontent.com/5842757/29248540-ea5362ae-801a-11e7-8f4a-94736b7cf7ea.png)

## Code:

https://github.com/mmd-osm/Overpass-API/commit/f030f214c64069e445a1aaabacb5b8a68ac94ced

**Comment by westnordost** [comment]
Say, could this then be used in place of an initial [bbox] for any kind of query? (Just asking, this is not a feature request)

**Comment by mmd-osm** [comment]
A global bbox `[bbox: ...]` is added implicitly to each single query statement as a constraint. That's not the case for `(around:...)`, which has to be added to each single query statement, like in the Road trip example above.

If you look how the around filter `(around:radius, lat, lon)` works today, this enhancement extends the idea from a single point to a linestring.

**Comment by mmd-osm** [comment]
Alternative implementation: https://github.com/drolbr/Overpass-API/issues/467#issuecomment-375462941

**Comment by drolbr** [comment]
The pull request has been merged. The feature is available in v0.7.55.

**Comment by michalfapso** [comment]
Thanks a lot for this feature! Here is my workflow for finding objects along a route:

1. Get route GPX file: https://map.project-osrm.org ("Export GPX File" button is in the bottom left panel)
2. Convert GPX to lat1,lon1,lat2,lon2,...
    cat route.gpx | tr '<' '\n' | grep '^trkpt lat.*lon' | sed 's/lat=//' | sed 's/lon=//' | sed 's/"//g' | sed 's/\/>//' | cut -d' ' -f2,3 | tr '\n' ' ' | tr ' ' ','
3. Go to http://overpass-turbo.eu/
Use query with
(around:radius,latitude_1,longitude_1,latitude_2,longitude_2,...,latitude_n,longitude_n)
More info here: https://github.com/drolbr/Overpass-API/issues/418
4. Example showing playgrounds within 500 meters of a short route through my town Piestany:
```
[out:json][timeout:25];
(
// node|way|rel
node[leisure=playground](around:500,48.60205,17.82415,48.60198,17.82423,48.60186,17.82436,48.60138,17.8249,48.6013,17.82498,48.60125,17.825,48.60125,17.825,48.6015,17.82555,48.60152,17.8256,48.60152,17.8256,48.60143,17.8257,48.60039,17.82688,48.60036,17.82692,48.60032,17.82697,48.60032,17.82697,48.60012,17.82692,48.59992,17.82685,48.59979,17.8268,48.5997,17.82676,48.5997,17.82676,48.5996,17.82669,48.5995,17.82662,48.59942,17.82654,48.59936,17.82648,48.5992,17.82628,48.59909,17.82615,48.59899,17.82605,48.59892,17.82598,48.59887,17.82594,48.59881,17.8259,48.59867,17.8258,48.59854,17.82573,48.59845,17.82568,48.59836,17.82565,48.59802,17.82551,48.59802,17.82551,48.598,17.82572,48.59805,17.82604,48.59806,17.82615,48.59806,17.82615,48.59805,17.82646,48.59806,17.8274,48.59807,17.82843,48.59807,17.82885,48.59804,17.8296,48.59807,17.82982,48.59811,17.82994,48.5982,17.83018,48.59847,17.83066,48.59855,17.83115,48.59855,17.83115,48.59836,17.83121,48.59793,17.83136,48.59714,17.83171,48.59643,17.83192,48.59607,17.83201,48.59595,17.83205,48.59573,17.83218,48.59559,17.83227,48.59525,17.83249,48.59497,17.83266,48.5947,17.83281,48.59384,17.83329,48.59356,17.83347,48.59333,17.83371,48.59302,17.83412,48.59276,17.8344,48.59262,17.83442,48.59262,17.83442,48.59262,17.83442,48.59249,17.83444,48.59246,17.83444,48.59246,17.83444,48.59244,17.83435,48.59241,17.83411,48.59235,17.83367,48.59233,17.83341,48.59236,17.83293,48.59236,17.83293,48.59234,17.83293,48.59159,17.83279,48.59155,17.83278,48.59144,17.83271,48.59117,17.83252,48.59114,17.8325,48.5911,17.83247,48.5911,17.83247,48.59106,17.83245,48.59102,17.83242,48.59078,17.83226,48.59056,17.83212,48.59037,17.83198,48.58981,17.83157,48.58974,17.83152,48.58974,17.83152,48.58962,17.83174,48.5893,17.83241,48.5893,17.83241,48.5893,17.83241,48.58915,17.83274,48.58893,17.83337,48.58889,17.83342,48.58888,17.8335,48.58885,17.8337,48.58885,17.8337,48.58882,17.83372,48.5888,17.83376,48.5888,17.83376,48.58865,17.83376,48.58827,17.83365,48.58824,17.83364,48.588,17.8336,48.58793,17.83359,48.58776,17.83354,48.58752,17.83344,48.58739,17.83339,48.58728,17.8333,48.58728,17.8333,48.58728,17.83327,48.58727,17.83324,48.58726,17.83321,48.58725,17.83319,48.58724,17.83317,48.58723,17.83315,48.58721,17.83313,48.5872,17.83312,48.58717,17.83311,48.58715,17.83311,48.58713,17.83311,48.5871,17.83312,48.58708,17.83314,48.58707,17.83315,48.58705,17.83318,48.58704,17.8332,48.58703,17.83324,48.58702,17.83328,48.58702,17.83332,48.58703,17.83336,48.58704,17.8334,48.58705,17.83343,48.58705,17.83343,48.587,17.83358,48.58699,17.83364,48.58699,17.83372,48.58699,17.8338,48.58698,17.83388,48.58698,17.83397,48.58707,17.83439,48.58724,17.83511,48.58731,17.83534,48.58744,17.83567,48.58774,17.83632,48.58788,17.83665,48.5881,17.83715,48.58812,17.8372,48.58817,17.83733,48.58822,17.83747,48.58825,17.83756,48.58825,17.83756,48.58838,17.83806,48.5884,17.83815,48.58848,17.83847,48.58855,17.83876,48.5886,17.83883,48.58864,17.83886,48.58867,17.83888,48.58872,17.83891,48.58872,17.83891);
);
// print results
out body;
>;
out skel qt;
```
